pkg/operator/bootstrapteardown/teardown.go

```diff
@@ -55,24 +58,32 @@ func TearDownBootstrap(config *rest.Config,
 		},
 	)
 
-	operatorSpec, _, _, err := operatorClient.GetOperatorState()
-	if err != nil {
-		return err
-	}
+	err := wait.PollInfinite(5*time.Second, func() (bool, error) {
+		etcd, err := etcdClient.Get("cluster", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
 
-	if err == nil {
 		klog.Infof("clusterversions is available, safe to remove bootstrap")
-		switch operatorSpec.ManagementState {
+		switch etcd.Spec.ManagementState {
 		case operatorv1.Managed:
 			if clusterMemberShipController.IsMember("etcd-bootstrap") {
-				return clusterMemberShipController.RemoveBootstrap()
+				//TODO: need to keep retry as long as we are sure bootstrap is not removed
+				err := clusterMemberShipController.RemoveBootstrap()
```

Comment:
nit `if err := clusterMemberShipController.RemoveBootstrap(); err != nil {`

scope of error is in this block only

Reply:
done